variables.tf

```diff
@@ -439,6 +459,21 @@ variable "enable_node_public_ip" {
   description = "(Optional) Should nodes in this Node Pool have a Public IP Address? Defaults to false."
 }
 
+variable "green_field_application_gateway_for_ingress" {
+  type = object({
+    name        = optional(string)
+    subnet_cidr = optional(string)
```

Comment:
For greenfield at least 1 of `subnet_id` or `subnet_cidr` should be provided. Should this be captured in a precondition ?